Modules/Core/Common/include/itkPoint.h

```diff
@@ -365,11 +365,11 @@ template <typename TValue, typename... TVariadic>
 auto
 MakePoint(const TValue firstValue, const TVariadic... otherValues)
 {
-  constexpr unsigned int dimension{ 1 + sizeof...(TVariadic) };
-
-  // Note that the class template argument deduction guide of std::array ensures that all values have the same type.
-  const std::array stdArray{ firstValue, otherValues... };
+  static_assert(std::conjunction_v<std::is_same<TVariadic, TValue>...>,
+                "The other values should have the same type as the first value.");
 
+  constexpr unsigned int              dimension{ 1 + sizeof...(TVariadic) };
+  const std::array<TValue, dimension> stdArray{ { firstValue, otherValues... } };
```

Comment:
AFAIK, `{{` `}}` are not required since C++14, 
`double-braces required in C++11 prior to the CWG 1270 revision (not needed in C++11 after the revision and in C++14 and beyond)`
(from https://en.cppreference.com/w/cpp/container/array, S. comment in the "Example")

Reply:
Thanks @issakomi Indeed the double-braces are not required anymore here. But they are certainly still allowed. It seems to me that they emphasize two things:
  - The initialization is not a copy-construction
  - The declaration does not use class template argument deduction (CTAD)

The original line of code did use CTAD _and_ did allow copy-construction (if `firstValue` would itself happen be an std::array, and there would be no _other values_). Which would only work with a single pair of braces:

    const std::array stdArray{ firstValue, otherValues... };

So the double-braces are still somewhat meaningful. What do you think?

Reply:
@issakomi In general, "wrapping" and copy-construction may be confused more easily when using single braces:

    int x = 1;
    std::array<int, 1> a{x}; // "wraps" x as array element
    std::array<int, 1> b{a}; // copy-construction

When using double-braces, copy-construction won't compile:

    std::array<int, 1> c{ {x} };  // OK, also wraps" x as array element
    std::array<int, 1> d{ {a} };  // Error: copy-construction fails to compile

See https://godbolt.org/z/GvcWq1Mx1

Having said so, I can understand if people find the double-braces a bit (too) noisy.

Reply:
Thank you! Sure, but, IMHO, in _particular case_ (template) it would be array or arrays, not a copy-construction, double/single braces do the same, s. [test](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:30,endLineNumber:11,positionColumn:30,positionLineNumber:11,selectionStartColumn:30,selectionStartLineNumber:11,startColumn:30,startLineNumber:11),source:'%23include+%3Carray%3E%0A%0Atemplate%3Ctypename+T,+typename...+Tv%3E%0Avoid+f(T+x,+Tv...+v)%0A%7B%0A++const+std::array%3CT,+sizeof...(Tv)+%2B+1%3E+a%7B+%7Bx,+v...%7D+%7D%3B%0A%7D%0A%0Aint+main()%0A%7B%0A++std::array%3Cint,+2%3E+a%7B42,+42%7D%3B%0A++f(a,+a,+a)%3B%0A++return+0%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:42.927110342840685,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g132,deviceViewOpen:'1',filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-std%3Dc%2B%2B17',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+13.2+(Editor+%231)',t:'0')),k:28.853837035975385,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+13.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+13.2+(Compiler+%231)',t:'0')),k:28.219052621183923,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).

Reply:
FYI, A new Clang compiler warning, `-Wctad-selects-copy`, is being proposed to avoid accidentally doing a copy-construction when _wrapping_ was intended, while using CTAD: [[Clang][Sema] Add -Wctad-selects-copy to diagnose copy deduction candidate](https://reviews.llvm.org/D155475) by PoYao Chang (@poyaoc97).

This new Clang warning could serve as an alternative to the existing `-Wctad-maybe-unsupported`, which produced undeserved warnings when using `std::lock_guard` or `std::scoped_lock`, as I reported at
- https://github.com/llvm/llvm-project/issues/63288